spec/operations/form/base_spec.rb

```diff
@@ -199,11 +250,87 @@
       end
     end
 
+    describe "#method_missing" do
+      let(:attributes) do
+        {
+          name: 42,
+          tags: ["tag1"]
+        }
+      end
+
+      specify { expect(form.name).to eq(42) }
+      specify { expect(form.tags).to eq(["tag1"]) }
+      specify { expect(form.build_author).to be_a(author_class) & have_attributes(title: nil) }
+
+      specify do
+        expect(form.build_author({ title: "foo" }, messages: {}))
+          .to be_a(author_class) & have_attributes(title: "foo")
+      end
+
+      specify { expect(form.build_post).to be_a(post_class) & have_attributes(id: nil, title: nil) }
+
+      specify do
+        expect(form.build_post({ title: "foo" }, messages: {}))
+          .to be_a(post_class) & have_attributes(id: nil, title: "foo")
+      end
+
+      specify { expect(form.build_tag).to be_nil }
+      specify { expect(form.foobar).to be_nil }
+    end
+
+    describe "#respond_to_missing?" do
+      specify { expect(form).to respond_to(:name) }
+      specify { expect(form).to respond_to(:tags) }
+      specify { expect(form).to respond_to(:build_author) }
+      specify { expect(form).to respond_to(:build_post) }
+      specify { expect(form).to respond_to(:author_attributes=) }
+      specify { expect(form).to respond_to(:posts_attributes=) }
+      specify { expect(form).not_to respond_to(:foobar) }
+      specify { expect(form).not_to respond_to(:build_tag) }
+      specify { expect(form).not_to respond_to(:tags_attributes=) }
+    end
+
     describe "#model_name" do
       specify { expect(form.model_name).to be_a(ActiveModel::Name) }
       specify { expect(form.model_name.to_s).to eq "Dummy::Form" }
     end
 
+    describe "#persisted?" do
+      subject(:persisted?) { form.persisted? }
+
+      it { is_expected.to be true }
+
+      context "when form has primary_key" do
```

Comment:
I was briefly browsing the setup, but couldn't see it - where is it actually specified in the setup that there is a primary key on this form?

Reply:
https://github.com/BookingSync/operations/pull/47/files#diff-015981d187060e2473315e3fed1c6480e985502fc8ef8c19c942b137ca0eafbdR25 here